### PR TITLE
[2.x] React update

### PR DESCRIPTION
**Pull request by EOstling**
This PR updates the example to use a functional react component. The reason behind the change is to follow the react community and encourage the use of functional components with the integration of react hooks, which have been the defacto use for react developers. The following example shows how intuitive it is to write react in this manner.
```
function Todo() {
  const [todo, setTodo] = useState(0);
  return (
    <div>
      <p>You have this many things todo:  {todo}</p>
      <button onClick={() => setTodo(todo + 1)}>
        Click me
      </button>
    </div>
  );
}

```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 composer.lock
 /phpunit.xml
 .phpunit.result.cache
+.idea
```
Comment (rodrigopedra) [inline .gitignore]: To be inline with `laravel/framework` .gitignore file, I would suggest you to add a forward slash in front of this folder name

`/.idea`

Reference: https://github.com/laravel/framework/blame/7.x/.gitignore

Comment (RichardStyles) [reply .gitignore]: I suggest this should be handled in your global gitignore rather than adding it here

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@
 composer.lock
 /phpunit.xml
 .phpunit.result.cache
+/.idea
```
Comment (driesvints) [inline .gitignore]: Remove this line please.

**Comment by driesvints** [comment]
Please try a new pr without al the idea files.

**Pull request closed without merging**